components/chat.tsx

```diff
@@ -26,26 +26,22 @@ export function Chat({ id, className, session, missingKeys }: ChatProps) {
   const [messages] = useUIState()
   const [aiState] = useAIState()
 
-  const [_, setNewChatId] = useLocalStorage('newChatId', id)
+  const [_, setNewChatAnimationId] = useLocalStorage('newChatAnimationId', id)
 
   useEffect(() => {
     if (session?.user) {
       if (!path.includes('chat') && messages.length === 1) {
         window.history.replaceState({}, '', `/chat/${id}`)
+        setNewChatAnimationId(id)
       }
     }
-  }, [id, path, session?.user, messages])
+  }, [id, path, session?.user, messages, setNewChatAnimationId])
 
   useEffect(() => {
-    const messagesLength = aiState.messages?.length
-    if (messagesLength === 2) {
+    if (aiState.refreshKey) {
```

Comment:
There will only be a refresh key at the end of a aiState.done(). No extra refreshes on page loads, but also will have refreshes at the end of each message, not just the first, fixing existing navigation issues like https://github.com/vercel/ai-chatbot/issues/364